src/app/home/layouts/budgets-form/budgets-form.component.ts

```diff
@@ -24,11 +31,26 @@ export class BudgetsFormComponent implements OnInit {
     this.initialSettingsService.sendCurrentStepInfo(3);
   }
 
+  selectCycle(cycle: CyclesDto) {
+    this.cycles.map(element => element.isSelected = false);
+    this.selectedCycle = cycle.name;
+    cycle.isSelected = true;
+  }
+
+  selectCategory(category: CategoryDto) {
+    this.categories.map(element => element.isSelected = false);
```

Comment:
Do not use map if you don't need the mapped output. For simple array mutation, just go with foreach.